src/game/system/_private/SoundSystem.cpp

```diff
@@ -34,19 +35,37 @@ void SoundSystem::Shutdown()
 
 void SoundSystem::Tick(deltaTime_t dt)
 {
+	//Updates sound location and velocity based on its TransformComponent position and PhysicsComponent speed
 	for (size_t i = 1; i < m_pSoundComponent.size(); ++i)
 	{
 		m_pSoundComponent[i]->Tick(dt);
 
-
-		//Updates sound location based on its TransformComponent position
-		m_pTransformComp = m_pSoundComponent[i]->getTransformComponent();
+		TransformComponent *pTransform = EntityManager::GetComponent<TransformComponent>(m_pSoundComponent[i]->GetEntity());
+		PhysicsComponent *pPhysics= EntityManager::GetComponent<PhysicsComponent>(m_pSoundComponent[i]->GetEntity());
 
 		std::vector<ALuint> sources = m_pSoundComponent[i]->getSource();
 
 		for (unsigned int j = 0; j < sources.size(); j++)
 		{
-			alSourcefv(sources[j], AL_POSITION, glm::value_ptr(m_pTransformComp->GetPosition()));
+			alSourcefv(sources[j], AL_POSITION, glm::value_ptr(pTransform->GetPosition()));
+			alSourcefv(sources[j], AL_VELOCITY, glm::value_ptr(pPhysics->GetVelocity()));
 		}
 	}
+
+	//Updating the listener position and orientation
+	ConstVector<CameraComponent *> m_pCameraComp = EntityManager::GetAll<CameraComponent>();
```

Comment:
This will iterate over all the CameraComponents and use the info from last one (which might not be the most recently created CameraComponent, depending on the order of creations and deletions of CameraComponents).
 
Buuut it's kind of tricky because only the DrawSystem knows which CameraComponent is the currently-used one. It might be better to refactor so that the CameraComponent class contains some static methods to get and set the current camera. The DrawSystem could then either get the current camera every tick, or the CameraComponent could have a static pointer to the current camera, and the DrawSystem could have a pointer pointer that references the CameraComponent's static pointer.